dart/lib/src/sentry_exception_factory.dart

```diff
@@ -6,34 +6,44 @@ import 'sentry_stack_trace_factory.dart';
 
 /// class to convert Dart Error and exception to SentryException
 class SentryExceptionFactory {
-  SentryStackTraceFactory _stacktraceFactory;
+  final SentryOptions _options;
+
+  final SentryStackTraceFactory _stacktraceFactory;
 
   SentryExceptionFactory({
-    SentryStackTraceFactory stacktraceFactory,
+    @required SentryStackTraceFactory stacktraceFactory,
     @required SentryOptions options,
-  }) {
-    if (options == null) {
+  })  : _options = options,
+        _stacktraceFactory = stacktraceFactory {
+    if (_options == null) {
       throw ArgumentError('SentryOptions is required.');
     }
 
-    _stacktraceFactory = stacktraceFactory ?? SentryStackTraceFactory(options);
+    if (_stacktraceFactory == null) {
+      throw ArgumentError('SentryStackTraceFactory is required.');
+    }
   }
 
   SentryException getSentryException(
     dynamic exception, {
     dynamic stackTrace,
     Mechanism mechanism,
   }) {
-    if (exception is Error) {
-      stackTrace ??= exception.stackTrace;
-    } else {
-      stackTrace ??= StackTrace.current;
+    SentryStackTrace sentryStackTrace;
+    if (stackTrace != null) {
+      sentryStackTrace ??= SentryStackTrace(
+        frames: _stacktraceFactory.getStackFrames(stackTrace),
+      );
+    } else if (exception is Error) {
+      sentryStackTrace ??= SentryStackTrace(
+        frames: _stacktraceFactory.getStackFrames(exception.stackTrace),
+      );
+    } else if (_options.attachStackTrace) {
+      sentryStackTrace ??= SentryStackTrace(
```

Comment:
why do we use `??=` if the var was just declared and its always null and `else if` branches

Reply:
✅